lux/lib/lux/prisms/discord/channel_prism.ex

```diff
@@ -0,0 +1,470 @@
+          if: %{properties: %{action: %{const: "set_permissions"}}},
+          then: %{required: ["channel_id", "overwrite_id", "allow", "deny"]}
+        },
+        # Channel type specific validations
+        %{
+          if: %{
+            properties: %{
+              type: %{enum: [2, 13]} # Voice channels
+            }
+          },
+          then: %{
+            properties: %{
+              topic: %{maxLength: 1024},
+              bitrate: %{type: :integer},
+              user_limit: %{type: :integer},
+              rtc_region: %{type: :string},
+              video_quality_mode: %{type: :integer}
+            }
+          }
+        },
+        %{
+          if: %{
+            properties: %{
+              type: %{enum: [15, 16]} # Forum/Media channels
+            }
+          },
+          then: %{
+            properties: %{
+              topic: %{maxLength: 4096},
+              available_tags: %{type: :array},
+              default_reaction_emoji: %{type: :object},
+              default_forum_layout: %{type: :integer},
+              default_sort_order: %{type: :integer}
+            }
+          }
+        }
+      ]
+    },
+    output_schema: %{
+      type: :object,
+      properties: %{
+        status: %{type: :string},
+        channel: %{
+          type: :object,
+          description: "Raw Discord API response"
+        }
+      },
+      required: ["status"]
+    }
+
+  require Logger
+
+  @base_url "https://discord.com/api/v10"
+  @timeout :timer.seconds(10)
+
+  @type api_response :: {:ok, map()} | {:error, map()}
+  @type validation_result :: {:ok, map()} | {:error, map()}
+  @type request_params :: %{
+    method: :get | :post | :put | :patch | :delete,
+    url: String.t(),
+    body: map() | nil
+  }
+
+  @doc """
+  Adds Discord bot token to request headers.
+  """
+  def add_auth_header(prism) do
+    token = Application.fetch_env!(:lux, :discord_token)
+    %{prism | headers: prism.headers ++ [{"Authorization", "Bot #{token}"}]}
+  end
+
+  @doc """
+  Handles the Discord API request with proper error handling and timeout management.
+  """
+  @impl true
+  def handler(input, ctx) do
+    with {:ok, request} <- prepare_request(input),
+         {:ok, response} <- execute_request(request) do
+      format_response(response)
+    end
+  end
+
+  @doc """
+  Prepares the API request based on validated parameters.
+  """
+  @spec prepare_request(map()) :: {:ok, request_params()} | {:error, map()}
+  def prepare_request(params) do
+    try do
+      request = build_request(params)
+      {:ok, request}
+    rescue
+      e in RuntimeError ->
+        Logger.error("Error preparing request: #{inspect(e)}")
+        {:error, %{type: "preparation_error", message: e.message}}
+    end
+  end
+
+  @doc """
+  Executes the API request asynchronously with timeout handling.
+  """
+  @spec execute_request(request_params()) :: api_response()
+  def execute_request(request) do
+    task = Task.async(fn ->
```

Comment:
Why do we need to do it async? We are waiting for it immediately after anyways so what is the point?